homeassistant/components/network/__init__.py

```diff
@@ -0,0 +1,88 @@
+"""The Network Configuration integration."""
+from __future__ import annotations
+
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components import websocket_api
+from homeassistant.core import HomeAssistant
+from homeassistant.loader import bind_hass
+
+from .const import (
+    ATTR_ADAPTERS,
+    ATTR_CONFIGURED_ADAPTERS,
+    DOMAIN,
+    NETWORK_CONFIG_SCHEMA,
+)
+from .models import Adapter
+from .network import Network
+
+ZEROCONF_DOMAIN = "zeroconf"  # cannot import from zeroconf due to circular dep
+_LOGGER = logging.getLogger(__name__)
+
+
+@bind_hass
+async def async_get_adapters(hass: HomeAssistant) -> list[Adapter]:
+    """Get the network adapter configuration."""
+    network: Network = hass.data[DOMAIN]
+    return network.adapters
+
+
+async def async_setup(hass: HomeAssistant, config: dict) -> bool:
+    """Set up network for Home Assistant."""
+
+    hass.data[DOMAIN] = network = Network(hass)
+    await network.async_setup()
+    await network.async_migrate_from_zeroconf(config.get(ZEROCONF_DOMAIN, {}))
+    network.async_configure()
+
+    _LOGGER.debug("Adapters: %s", network.adapters)
+
+    websocket_api.async_register_command(hass, websocket_network_adapters)
+    websocket_api.async_register_command(hass, websocket_network_adapters_configure)
+
+    return True
+
+
+@websocket_api.require_admin  # type: ignore[arg-type]
+@websocket_api.async_response
+@websocket_api.websocket_command({vol.Required("type"): "network"})
```

Comment:
This might work.
```suggestion
@websocket_api.async_response
@websocket_api.require_admin
@websocket_api.websocket_command({vol.Required("type"): "network"})
```

Reply:
no dice on that one

Reply:
```homeassistant/components/network/__init__.py:48: error: Argument 1 has incompatible type "Callable[[HomeAssistant, ActiveConnection, Dict[Any, Any]], Coroutine[Any, Any, None]]"; expected "Callable[[HomeAssistant, ActiveConnection, Dict[Any, Any]], None]"  [arg-type]
homeassistant/components/network/__init__.py:48: error: Argument 1 to "async_response" has incompatible type "Callable[[HomeAssistant, ActiveConnection, Dict[Any, Any]], None]"; expected "Callable[[HomeAssistant, ActiveConnection, Dict[Any, Any]], Awaitable[None]]"  [arg-type]
homeassistant/components/network/__init__.py:67: error: Argument 1 has incompatible type "Callable[[HomeAssistant, ActiveConnection, Dict[Any, Any]], Coroutine[Any, Any, None]]"; expected "Callable[[HomeAssistant, ActiveConnection, Dict[Any, Any]], None]"  [arg-type]
homeassistant/components/network/__init__.py:67: error: Argument 1 to "async_response" has incompatible type "Callable[[HomeAssistant, ActiveConnection, Dict[Any, Any]], None]"; expected "Callable[[HomeAssistant, ActiveConnection, Dict[Any, Any]], Awaitable[None]]"  [arg-type]```

Reply:
Thanks for checking 🙂